web/src/engine/websites/ArgosScan.ts

```diff
@@ -0,0 +1,180 @@
+import { Tags } from '../Tags';
+import icon from './ArgosScan.webp';
+import { Chapter, DecoratableMangaScraper, Manga, Page, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchGraphQL, FetchRequest } from '../FetchProvider';
+
+type APIMangas = {
+    getProjects: {
+        projects: {
+            id: number,
+            name: string
+        }[]
+    }
+};
+
+type APIManga = {
+    project: {
+        name: string,
+        id: number,
+        getChapters: {
+            id: number,
+            number: number,
+            title: string
+        }[]
+    }
+}
+
+type APIChapter = {
+    getChapters: {
+        chapters: {
+            id: number,
+            number: number,
+            title: string
+            images: string[]
+        }[]
+    }
+}
+
+@Common.ImageAjax()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('argosscan', `Argos Scan`, 'https://argosscan.com', Tags.Language.Portuguese, Tags.Source.Scanlator, Tags.Media.Manhwa, Tags.Media.Manga);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return /https?:\/\/argosscan\.com\/obras\/([\d]+)/.test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const id = parseInt(new URL(url).pathname.match(/obras\/(\d+)\//i)[1]);
+        const gql = `
+            query project($id: Int!) {
+                project(id: $id) {
+                    name
+                }
+            }
+        `;
+        const vars = { id: id };
+        const request = new FetchRequest(new URL('/graphql', this.URI).href);
+        const data = await FetchGraphQL<APIManga>(request, 'project', gql, JSON.stringify(vars));
+        const title = data.project.name;
+        return new Manga(this, provider, String(id), title.trim());
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList = [];
+        for (let page = 1, run = true; run; page++) {
+            const mangas = await this._getMangasFromPage(page, provider);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
+
+    async _getMangasFromPage(page: number, provider: MangaPlugin): Promise<Manga[]> {
+        const gql = `
+            query latestProjects($filters: FiltersExpression!, $orders: OrdersExpression!, $pagination: PaginationInput) {
+                getProjects(orders: $orders, filters: $filters, pagination: $pagination) {
+                    projects {
+                        id, name
+                    } 
+                }
+            }
+        `;
+        const vars = {
```

Comment:
The double quotes on JS object keys are not required for keys without certain special characters and can be dismissed